File: Backend/.NET/TicketManagementSystem_NET_Backend/TicketManagementSystem_NET_Backend/Security/MyCustomSecurity.cs
using TicketManagementSystem_NET_Backend.Repositories.UsersRepository;
using TicketManagementSystem_NET_Backend.Utils;

namespace TicketManagementSystem_NET_Backend.Security
{
    public enum UserRoles
    {
        CUSTOMER,
        ADMIN
    };
    public class MyCustomSecurity
    {
        private static IUsersRepository _usersRepository;

        public MyCustomSecurity()
        {
        }

        public static async void ValidateRequest(HttpContext context, UserRoles userRole)
        {
            _usersRepository = new UsersRepository();
            if (!RequestContainsAuthorization(context.Request))
            {
                throw new NoAuthorizationException("No authorization header in request!");
            }

            string jwt = ExtractJWT(context.Request);
            string email = JwtUtil.ExtractEmail(jwt);
            bool isCustomer = await GetUserRole(email);

            if (isCustomer && userRole != UserRoles.CUSTOMER)
            {
                throw new ResourceForbiddenException("Resource not allowed!");
            }

            if (!isCustomer && userRole != UserRoles.ADMIN)
            {
                throw new ResourceForbiddenException("Resource not allowed!");
            }
        }

        public static bool RequestContainsAuthorization(HttpRequest request)
        {
            return !string.IsNullOrEmpty(request.Headers["Authorization"]);
        }

        public static string ExtractJWT(HttpRequest request)
        {
            string authorizationHeader = request.Headers["Authorization"];

            if (!authorizationHeader.StartsWith("Bearer"))
            {
                throw new NoBearerInAuthorizationException("Authorization header not starting with Bearer!");
            }

            return authorizationHeader.Substring(7);
        }

        public async static Task<bool> GetUserRole(string email)
        {
            return await _usersRepository.FindRoleByEmail(email);
        }
    }
}

File: Backend/Flask/security/MyCustomSecurity.py
from JwtUtil import JwtUtil
from data.UsersRepository import UsersRepository
class UserRoles:
    CUSTOMER = "CUSTOMER"
    ADMIN = "ADMIN"

class NoAuthorizationException(Exception):
    pass
class ResourceForbiddenException(Exception):
    pass

class MyCustomSecurity:
    users_repository = UsersRepository()

    @staticmethod
    def validate_request(context, user_role):
        if not MyCustomSecurity.request_contains_authorization(context):
            raise NoAuthorizationException("No authorization header in request!")

        jwt_token = MyCustomSecurity.extract_jwt(context)
        email = JwtUtil.extract_email(jwt_token)
        is_customer = MyCustomSecurity.get_user_role(email)
        if is_customer and user_role != UserRoles.CUSTOMER:
            raise ResourceForbiddenException("Resource not allowed!")

        if not is_customer and user_role != UserRoles.ADMIN:
            raise ResourceForbiddenException("Resource not allowed!")

    @staticmethod
    def request_contains_authorization(request):
        return "Authorization" in request.headers

    @staticmethod
    def extract_jwt(request):
        authorization_header = request.headers["Authorization"]

        if not authorization_header.startswith("Bearer"):
            raise NoAuthorizationException("Authorization header not starting with Bearer!")

        return authorization_header[7:]

    @staticmethod
    def get_user_role(email):
        return MyCustomSecurity.users_repository.find_role_by_email(email)
